authz/grpc_audit_end2end_test.go

```diff
@@ -0,0 +1,304 @@
+package authz
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"io"
+	"net"
+	"os"
+	"testing"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/authz/audit"
+	"google.golang.org/grpc/credentials"
+	testgrpc "google.golang.org/grpc/interop/grpc_testing"
+	testpb "google.golang.org/grpc/interop/grpc_testing"
+	"google.golang.org/grpc/testdata"
+
+	_ "google.golang.org/grpc/authz/audit/stdout"
+)
+
+type testServer struct {
+	testgrpc.UnimplementedTestServiceServer
+}
+
+func (s *testServer) UnaryCall(ctx context.Context, req *testpb.SimpleRequest) (*testpb.SimpleResponse, error) {
+	return &testpb.SimpleResponse{}, nil
+}
+
+type statAuditLogger struct {
+	Stat         map[bool]int
+	stdoutLogger audit.Logger
+	SpiffeIds    []string
```

Comment:
We're checking all the fields during unit testing so I don't think we need to duplicate it here. I try to concentrate on more global checks for this test. To be honest, I don't even think that SPIFFE belongs here but since there is a lack of tests for it I included it.

Reply:
Technically I understand they don't belong here but I still think it's beneficial to verify all the fields since we mocked them in our unit tests. We would thank ourselves if some change in the future altered the behavior and the was the only failing test.

We don't have to be that exhaustive so if you could at least verify all fields in one test case I think that'd be decent.